File: frontend/src/types/gpsBreadcrumb.ts
export interface GpsBreadcrumb {
  id: number;
  trip_id: string;
  vehicle_id: string | null;
  timestamp: string;
  lat: number | null;
  lon: number | null;
  speed_kmph: number | null;
  heading_deg: number | null;
}

File: backend/app/schemas/gps_breadcrumb.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class GpsBreadcrumbRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    timestamp: datetime
    lat: float | None
    lon: float | None
    speed_kmph: float | None
    heading_deg: int | None
